### DISCUSSION TITLE
Map geometry isn't visible in Unity even it's there.

### DISCUSSION THREAD

**Discussion by ExacoMvm** [AssetRipper/AssetRipper#1003] [General]
Hey there, does anyone know how to fix this issue ? The geometry and textures seem to be fine but for some reason I just can't see the geometry in Editor Viewport, except in Wireframe mode and few others.

I think it might be just Unity thing and not AssetRipper issue, some smaller maps of same game work fine(ish) tho.

**Comment by ds5678** [comment]
If you're using static mesh separation, that's probably why. It's an incomplete feature.

**Reply by ExacoMvm** [reply]
I didn't use it, only used to skip StreamingAssets and changed few things like to skip exporting scripts, terrain set to 3D Mesh. Even tho before I've tried different settings including default and it's same thing every time.

After trying to import the assets folder directly to Unity rather than copying to the project files and then running Unity I think it started to show more meshes, but some are still missing or there's some w/ missing textures. Could it be that some of those assets are in the StreamingAssets and that's why I don't get everything ripped ?

The game is EFT btw. "Factory" and "Hideout" are probably the only maps that are working nearly perfectly, it's the smallest maps of the game.

**Reply by ds5678** [reply]
No idea

**Reply by Prosopho** [reply]
Probably just invalid shaders not rendering things since you're talking about tarkov. On some maps like interchange, customs, reserve etc. Some objects are set as inactive by default in the scenes. But if you don't have valid shaders, then yeah nothing won't render. (Assuming that the gameobjects are set as active.)

**Comment by Zbuffers** [comment]
Hi, 
also could be a material/shader/UV issue...
Try assigning default material(s) with no texture, just to check
static mesh separation unfortunately breaks the UVs...
my 2 cents ;)
Z

**Comment by dukemagus** [comment]
Don't skip streaming assets

Em sex., 25 de ago. de 2023 10:43, Jeremy Pritts ***@***.***>
escreveu:

**Reply by ExacoMvm** [reply]
Tried it, makes no difference, the maps lack exact same models/textures.
Maybe I'm just using the wrong settings ( which is default ).

This is example of full interchange imported and there's missing like 99% of the assets.
https://i.gyazo.com/40afe5f897b8246686e99becee4f17c6.png

**Reply by Prosopho** [reply]
The game objects for the shelves and stuff are set as inactive so you have to enable them for them to render.

**Reply by ExacoMvm** [reply]
Thanks, I've found that in most of those objects "Mesh Renderer" is disabled for some reason, now I only need to figure out how to enable them all the quick way.

**Reply by ds5678** [reply]
Oh, the game developer must have disabled them.

**Reply by maxpayne88zz-hub** [reply]
> Thanks, I've found that in most of those objects "Mesh Renderer" is disabled for some reason, now I only need to figure out how to enable them all the quick way.

Hi did you figure out how to fix it BTW?i have some items on the resrve map that look like mesh from far away till i get up close then the item fully renders.
